src/plugins/stake-plugins/generic/policyAdapters/TarotPoolAdaptor.ts

```diff
@@ -388,10 +391,10 @@ export const makeTarotPoolAdapter = (policyConfig: StakePolicyConfig<TarotPoolAd
         }
         return await tarotRouterContract.populateTransaction.leverage(
           poolContractAddress,
-          token0Amount,
-          token1Amount,
-          amountToken0Min,
-          amountToken1Min,
+          token0Amount.mul(leverage - 1),
+          token1Amount.mul(leverage - 1),
+          amountToken0Min.mul(leverage - 1),
+          amountToken1Min.mul(leverage - 1),
```

Comment:
Why 1 less?

Reply:
What if leverage is 1?

Reply:
1x is unleveraged which is where you're at after calling mintCollateral since you haven't borrowed anything yet. To get Nx leverage you need to borrow (N - 1) * deposit amount

Reply:
Makes sense. But we have a problem with 1 - 1 = 0 (and we're dividing by zero). I think BigNumber might crap out on divide by 0.

Reply:
Put a runtime check at the top of the adapter to error on leverage <= 1.